pallets/metaverse/src/lib.rs

```diff
@@ -285,11 +556,66 @@ impl<T: Config> Pallet<T> {
 	pub fn account_id() -> T::AccountId {
 		T::MetaverseTreasury::get().into_account()
 	}
+
+	/// Update staking info of origin
+	fn update_staking_info(who: &T::AccountId, staking_info: BalanceOf<T>) {
+		if staking_info.is_zero() {
+			StakingInfo::<T>::remove(&who);
+			T::Currency::remove_lock(LOCK_STAKING, &who);
+		} else {
+			T::Currency::set_lock(LOCK_STAKING, &who, staking_info, WithdrawReasons::all());
+			StakingInfo::<T>::insert(who, staking_info);
+		}
+	}
+
+	/// This method will be exposed to mining trait to be executed to pay staker reward per round
+	fn pay_staker(
+		metaverse_id: MetaverseId,
+		round: RoundIndex,
+		total_reward: BalanceOf<T>,
+	) -> DispatchResultWithPostInfo {
+		// Get staking info of metaverse and current round
+		let mut metaverse_stake_per_round: MetaverseStakingPoints<T::AccountId, BalanceOf<T>> =
+			Self::get_metaverse_stake_per_round(&metaverse_id, round)
+				.ok_or(Error::<T>::MetaverseStakingInfoNotFound)?;
+
+		ensure!(
+			metaverse_stake_per_round.claimed_rewards.is_zero(),
+			Error::<T>::MetaverseStakingAlreadyPaid
+		);
+
+		ensure!(
+			!metaverse_stake_per_round.stakers.is_empty(),
+			Error::<T>::MetaverseHasNoStake
+		);
+
+		let metaverse_staking_snapshot_by_round: MetaverseStakingSnapshot<BalanceOf<T>> =
```

Comment:
metaverse_staking_snapshot_by_round not being used anywhere

Reply:
That's right, was thinking of using a network snapshot but may not need it for now.